### PR TITLE
WALMA-133: Adding fake video capture file configuration to chrome

### PR DESCRIPTION
**Pull request by dministro**
[WALMA-133](https://lombiq.atlassian.net/browse/WALMA-133)

[WALMA-133]: https://lombiq.atlassian.net/browse/WALMA-133?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.Tests.UI/Constants/DirectoryPaths.cs [hunk 1]
```diff
@@ -10,8 +10,11 @@ public static class DirectoryPaths
     public const string Temp = nameof(Temp);
     public const string Screenshots = nameof(Screenshots);
 
+    public static string GetTempDirectoryPath(params string[] subDirectoryNames) =>
+        Path.Combine(new[] { Environment.CurrentDirectory, Temp }.Concat(subDirectoryNames).ToArray());
+
     public static string GetTempSubDirectoryPath(string contextId, params string[] subDirectoryNames) =>
-        Path.Combine(new[] { Environment.CurrentDirectory, Temp, contextId }.Concat(subDirectoryNames).ToArray());
+        GetTempDirectoryPath(new[] { contextId }.Concat(subDirectoryNames).ToArray());
```
Comment (0liver) [inline Lombiq.Tests.UI/Constants/DirectoryPaths.cs]: I would argue, that as long as you don't reuse the `GetTempDirectoryPath` method anywhere, it's overhead to create two new arrays just to concat file paths.

Instead, you can stack `Path.Combine` calls:

```suggestion
        Path.Combine(Path.Combine(Environment.CurrentDirectory, Temp, contextId), Path.Combine(subDirectoryNames));
```

Comment (0liver) [reply Lombiq.Tests.UI/Constants/DirectoryPaths.cs]: Maybe formatted like this:
```suggestion
        Path.Combine(
            Path.Combine(Environment.CurrentDirectory, Temp, contextId),
            Path.Combine(subDirectoryNames));
```

Comment (dministro) [reply Lombiq.Tests.UI/Constants/DirectoryPaths.cs]: `GetTempDirectoryPath` is in use here: https://github.com/Lombiq/UI-Testing-Toolbox/blob/c69a65bd0906fd4f89236eb1d964b84f060a6b3f/Lombiq.Tests.UI/Extensions/FakeBrowserVideoSourceExtensions.cs#L14

But you are right stacking `Path.Combine` looks better.

File: Lombiq.Tests.UI/Docs/FakeVideoCaptureSource.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Fake video capture source
+
+## Preparing video file
+
+You can use video files as a fake video capture source in Chrome browser of format `y4m` or `mjpeg`.
+
+If you have a video file in eg.: `mp4` format use your preferred video tool to convert it to one of the formats mentioned above. If you don't have a preferred tool, simply use `ffmpeg`.
```
Comment (Psichorex) [inline Lombiq.Tests.UI/Docs/FakeVideoCaptureSource.md]:
```suggestion
If you have a video file in e.g., `mp4` format, use your preferred video tool to convert it to one of the formats mentioned above. If you don't have a preferred tool, simply use `ffmpeg`.
```

File: Lombiq.Tests.UI/Docs/FakeVideoCaptureSource.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ffmpeg -y -i test.mp4 test.mjpeg
+
+# Convert with resize to 480p.
+ffmpeg -y -i test.mp4 -filter:v scale=480:-1 test.mjpeg
+
+```
+
+### Configuring test
+
+Add the prepared video file to your `csproj` as an `EmbeddedResource`, then configure the test method like below.
```
Comment (Psichorex) [inline Lombiq.Tests.UI/Docs/FakeVideoCaptureSource.md]:
```suggestion
Add the prepared video file to your `csproj` project as an `EmbeddedResource`, and then configure the test method as shown below.
```

File: Lombiq.Tests.UI/Models/FakeBrowserVideoSource.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+using System;
+using System.IO;
+
+namespace Lombiq.Tests.UI.Models;
+
+public class FakeBrowserVideoSource
+{
+    /// <summary>
+    /// Gets or sets a callback that'll be used to get the video content <see cref="Stream"/> and will be saved and used
+    /// as a fake video capture file for the chrome browser. The consumer will dispose the returned <see cref="Stream"/>
```
Comment (Psichorex) [inline Lombiq.Tests.UI/Models/FakeBrowserVideoSource.cs]:
```suggestion
    /// as a fake video capture file for the Chrome browser. The consumer will dispose of  the returned <see cref="Stream"/>
```

File: Lombiq.Tests.UI/Models/FakeBrowserVideoSource.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    /// after the callback gets called.
```
Comment (Psichorex) [inline Lombiq.Tests.UI/Models/FakeBrowserVideoSource.cs]:
```suggestion
    /// after the callback is called.
```

File: Lombiq.Tests.UI/Models/FakeBrowserVideoSource.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+using System;
+using System.IO;
+
+namespace Lombiq.Tests.UI.Models;
+
+public class FakeBrowserVideoSource
+{
+    /// <summary>
+    /// Gets or sets a callback that'll be used to get the video content <see cref="Stream"/> and will be saved and used
```
Comment (Psichorex) [inline Lombiq.Tests.UI/Models/FakeBrowserVideoSource.cs]:
```suggestion
    /// Gets or sets a callback that will be used to obtain the video content <see cref="Stream"/> and will be saved and used
```

**Review by Psichorex — CHANGES_REQUESTED** [review]

File: Lombiq.Tests.UI/Constants/DirectoryPaths.cs (hunk 1 above)
Comment (Psichorex) [reply Lombiq.Tests.UI/Constants/DirectoryPaths.cs]: @0liver Please, if you consider this suggestion you have made here resolved, make the thread resolved aswell.

**Pull request merged**